jpo-geojsonconverter/src/main/java/us/dot/its/jpo/geojsonconverter/StreamsExceptionHandler.java

```diff
@@ -0,0 +1,45 @@
+package us.dot.its.jpo.geojsonconverter;
+
+import org.apache.kafka.streams.errors.StreamsUncaughtExceptionHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+
+/**
+ * Handler for unhandled exceptions thrown from Streams topologies that
+ *  logs the exception to a topic, and allows choosing the shutdown behavior.
+ * 
+ * See {@link https://cwiki.apache.org/confluence/display/KAFKA/KIP-671%3A+Introduce+Kafka+Streams+Specific+Uncaught+Exception+Handler}
+ * for a description of the options.
+ */
+public class StreamsExceptionHandler implements StreamsUncaughtExceptionHandler {
+
+    final static Logger logger = LoggerFactory.getLogger(StreamsExceptionHandler.class);
+
+    final String topology;
+    
+    @Autowired
+    public StreamsExceptionHandler(String topology) {
+        this.topology = topology;
+    }
+    
+    @Override
+    public StreamThreadExceptionResponse handle(Throwable exception) {
+
+        logger.error(String.format("Uncaught exception in stream topology %s", topology), exception);
```

Comment:
Consider sending this category of error to a topic in addition to logging so that it can trigger alerts?

Reply:
We could do that, @drewjj what do you think would make the most sense here?

Reply:
We could do that, @drewjj what do you think would make the most sense here?

Reply:
That would be fine, I think we were planning to do something along those lines in the past with the schema validations but we ended up combining those with the messages themselves. We definitely want to know when this fails so it would be good to do something like this. This would allow us to have a built in alert within the deployment instead of needing to have a log listener like prometheus doing it.